src/test/scala/org/broadinstitute/dsde/workbench/sam/service/TosServiceSpec.scala

```diff
@@ -29,82 +33,137 @@ class TosServiceSpec extends AnyFlatSpec with TestSupport with BeforeAndAfterAll
+    adherenceStatusNotAcceptedLatest.userHasAcceptedLatestTos shouldBe false
+    adherenceStatusNotAcceptedLatest.acceptedTosAllowsUsage shouldBe false
+
+    tosServiceV2.acceptTosStatus(userAcceptedV0.id, samRequestContext).unsafeRunSync() shouldBe true
+    val userAcceptedV2 = dirDAO.loadUser(userAcceptedV0.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatusV2 = tosServiceV2.getTosAdherenceStatus(userAcceptedV2).unsafeRunSync()
+    adherenceStatusV2.userHasAcceptedLatestTos shouldBe true
+    adherenceStatusV2.acceptedTosAllowsUsage shouldBe true
   }
 
-  it should "not allow users who have never accepted a ToS version to use the API, even with a grace period enabled" in {
+  it should "always allow service account users to use Terra" in {
     assume(databaseEnabled, databaseEnabledClue)
-    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
     val tosServiceGracePeriodEnabled =
-      new TosService(dirDAO, "example.com", TestSupport.tosConfig.copy(enabled = true, version = "2", isGracePeriodEnabled = true))
-    tosServiceGracePeriodEnabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(false)
+      new TosService(mockDirDAO, TestSupport.tosConfig.copy(version = "2"))
+    val adherenceStatus = tosServiceGracePeriodEnabled.getTosAdherenceStatus(serviceAccountUser).unsafeRunSync()
+    adherenceStatus.userHasAcceptedLatestTos shouldBe false
+    adherenceStatus.acceptedTosAllowsUsage shouldBe true
   }
 
-  it should "allow users that have accepted a previous ToS version to use the API when the grace period is enabled" in {
-    assume(databaseEnabled, databaseEnabledClue)
-    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
-    dirDAO.acceptTermsOfService(defaultUser.id, "1", samRequestContext).unsafeRunSync()
-    val userAcceptedPreviousVersion = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+  /** | Case | Grace Period Enabled | Accepted Version | Current Version | User accepted latest | User Can Use Terra |
+    * |:-----|:---------------------|:-----------------|:----------------|:---------------------|:-------------------|
+    * | 1    | false                | null             | "2"             | false                | false              |
+    * | 2    | false                | "0"              | "2"             | false                | false              |
+    * | 3    | false                | "2"              | "2"             | true                 | true               |
+    * | 4    | true                 | null             | "2"             | false                | false              |
+    * | 5    | true                 | "0"              | "2"             | false                | true               |
+    * | 6    | true                 | "2"              | "2"             | true                 | true               |
+    */
```

Comment:
Might want to consider breaking out the "Adherence Status" checks into their own close just for readability and such.  I do not feel strongly about this though, just as files get bigger, they can cause their own challenges and problems if not careful.